application/helpers/admin/import_helper.php

```diff
@@ -1410,6 +1413,28 @@ function XMLImportSurvey($sFullFilePath, $sXMLdata = null, $sNewSurveyName = nul
             }
         }
 
+        // Email attachments are with relative paths on the file, but are currently expected to be saved as absolute.
+        // Transforming them from relative paths to absolute paths.
+        if (!empty($insertdata['attachments'])) {
+            // NOTE: Older LSS files have attachments as a serialized array, while newer ones have it as a JSON string.
+            // Serialized attachments are not supported anymore.
+            $attachments = json_decode($insertdata['attachments'], true);
+            if (!empty($attachments) && is_array($attachments)) {
+                $uploadDir = realpath(Yii::app()->getConfig('uploaddir'));
+                foreach ($attachments as &$template) {
+                    foreach ($template as &$attachment) {
+                        if (!isAbsolutePath($attachment['url'])) {
+                            $attachment['url'] = $uploadDir . DIRECTORY_SEPARATOR . $attachment['url'];
+                        }
+                    }
+                }
+            } elseif (is_null($attachments)) {
+                // JSON decode failed. Most probably the attachments were in the PHP serialization format.
+                $wrongAttachmentsFormat = true;
+            }
+            $insertdata['attachments'] = serialize($attachments);
```

Comment:
This line should move into the if statement above. We don't want to serialise if we did not manage to parse anything from the file.

Reply:
You meant to avoud serializing an empty string?
I believe the rest of the code expects serlized data there. Even if empty. This is working on master.

Reply:
If we dont parse anything it should be an empty string. Right now if we dont parse anything we might store invalid data.

Reply:
> If we dont parse anything it should be an empty string

Or a seralized empty string?

Reply:
Serialised empty string is fine. Right now it's serialising anything coming from the input which means I could enter serialised PHP in the .lss file and it would serialise that. Maybe that wont break anything but do we want that?

Reply:
Sure its a step in the right direction.